src/Stormpath/Http/HttpClientRequestExecutor.php

```diff
@@ -22,17 +22,26 @@
 use Guzzle\Http\Client;
 use Guzzle\Http\Message\RequestInterface;
 use Stormpath\ApiKey;
+use Stormpath\Http\Authc\RequestSigner;
 use Stormpath\Http\Authc\Sauthc1Signer;
+use Stormpath\Stormpath;
 
 class HttpClientRequestExecutor implements RequestExecutor
 {
+    private $authenticationScheme = Stormpath::SAUTHC1_AUTHENTICATION_SCHEME;
     private $httpClient;
     private $signer;
 
-    public function __construct()
+    public function __construct(RequestSigner $signer = null)
     {
         $this->httpClient = new Client();
-        $this->signer = new Sauthc1Signer;
+
+        if (!$signer) {
+            $signer = "\\Stormpath\\Http\\Authc\\" . $this->authenticationScheme . "RequestSigner";
```

Comment:
I don't think you need to use reflection here. Just create a new instance of `Sauthc1RequestSigner`.